src/common/utility.ts

```diff
@@ -92,6 +95,78 @@ export class Utility {
         return mysql.createConnection(newConnectionOptions);
     }
 
+    public static async editConnection(connectionNode: ConnectionNode, context: vscode.ExtensionContext, mysqlTreeDataProvider: MySQLTreeDataProvider) {
+        if (connectionNode) {
+            connectionNode.editConnection(context, mysqlTreeDataProvider);
+        } else {
+            // No connection has been selected, let the user pick one
+            const connections = context.globalState.get<{ [key: string]: IConnection }>(Constants.GlobalStateMySQLConectionsKey);
+            const items: vscode.QuickPickItem[] = [];
+
+            if (connections) {
+                for (const id of Object.keys(connections)) {
+                    const item = connections[id];
+                    items.push({
+                      label: item.host,
+                      description: item.user});
+                }
+            }
+
+            vscode.window.showQuickPick(items).then(async (selection) => {
+                // the user canceled the selection
+                if (!selection) {
+                    return;
+                }
+
+                const selectedConnectionId = Object.keys(connections)[items.indexOf(selection)];
+                const selectedConnection = connections[selectedConnectionId];
+
+                if (selectedConnection !== undefined) {
+                    const dummyNode = new ConnectionNode(selectedConnectionId, selectedConnection.host, selectedConnection.user, selectedConnection.password,

```

Comment:
Is it good practice to do this?

Reply:
Maybe not.. but should we move the edit/delete methods away from the connectionNode and onto the connection object instead? 
We could change the IConnection to an abstract class instead, and implement the methods in here instead?